### ISSUE TITLE
Bug introduced from 1.1.4 and later

### ISSUE THREAD

**Issue by ravjsdev** [vuejs/vue-test-utils#1845]
### Subject of the issue

My unit tests suddenly started breaking from 1.1.4 and later

### Steps to reproduce

since updating to 1.1.4 (or greater) my unit tests are now breaking - I have a custom component (renderer) which renders form components (e.g. input, text area, checkbox etc) dynamically using a config provided via props. The component iterates through the config and renders the inputs (mapping them to appropriate vue components) - now in my test when i do the following:

```
...
const dynamicFormConfig = ['text-area', 'text-input'];

const options = merge({
    propsData: {
      dynamicFormConfig,
    },
    store,
    localVue,
  }, params.options);

textInput = mount(RendererComponent, options).findComponent({` name: 'MyTextInput' });

expect(textInput.exists()).toBe(true); 
```
i should get true, but now my tests fail with false.

### Expected behaviour

I should get true from the test expectation but now im getting false. When i downgrade to 1.1.3 everything works fine and the tests pass.

### Actual behaviour

The test now fails. (i.e. textInput.exist() returns false)

### Possible Solution

Revert the breaking change introduced in 1.1.4 (Djaler:fix/find-functional-component)

**Comment by lmiller1990** [comment]
I don't think we should revert a change that fixes a real bug. Let's just fix this new bug. Can you please post your entire test and component? We can add it to the test suite, then make it pass. Then both bugs will be fixed. What do you think? 👍

**Comment by ravjsdev** [comment]
ok sure i'll put a demo repo together which replicates this problem

**Comment by lmiller1990** [comment]
Thank you!

**Comment by GeorgieH** [comment]
I've worked with ravjsdev to create a minimal example which reproduces the issue. Here it is: https://github.com/GeorgieH/vue-test-utils-1845

Info is in the README. Let me know if you have any questions.

**Comment by lmiller1990** [comment]
Thanks.

**Comment by lmiller1990** [comment]
Released in 1.2.1 https://github.com/vuejs/vue-test-utils/releases/tag/v1.2.1.

This should be good now, if not lmk and I will re-open this.
